hera_qm/xrfi.py

```diff
@@ -299,4 +301,51 @@ def xrfi_run(files, opts, history):
         else:
             raise ValueError('Unrecognized output file format ' + str(opts.outfile_format))
 
+        if opts.summary:
+            sum_file = ''.join([basename, opts.summary_ext])
+            sum_path = os.path.join(dirname, sum_file)
+            summarize_flags(uvd, sum_path)
     return
+
+
+def summarize_flags(uv, outfile):
+    """ Collapse several dimensions of a UVData flag array to summarize.
+    Args:
+        uv -- UVData object containing flag_array to be summarized
+        outfile -- filename for output npz file
+
+    Return:
+        npz file containing "spectrum" of max, min, mean, std, median of flags
+    """
+    import pyuvdata.utils as uvutils
+
+    # Average across bls for given time
+    waterfall = np.zeros((uv.Ntimes, uv.Nfreqs, uv.Npols))
+    waterfall_weight = np.zeros(uv.Ntimes)
+    times = np.unique(uv.time_array)
+    for ti, time in enumerate(times):
+        ind = np.where(uv.time_array == time)
+        waterfall_weight[ti] = len(ind)
+        waterfall[ti, :, :] = np.mean(uv.flag_array[ind, 0, :, :])
+    # Calculate stats across time
+    tmax = np.max(waterfall, axis=0)
+    tmin = np.min(waterfall, axis=0)
+    tmean = np.average(waterfall, axis=0, weights=waterfall_weight)
+    tstd = np.sqrt(np.average((waterfall - tmean[np.newaxis, :, :])**2,
+                              axis=0, weights=waterfall_weight))
+    tmedian = np.median(waterfall, axis=0)
+    # Calculate stats across frequency
+    fmax = np.max(waterfall, axis=1)
+    fmin = np.min(waterfall, axis=1)
+    fmean = np.mean(waterfall, axis=1)  # no weights - all bls have all channels
+    fstd = np.std(waterfall, axis=1)
+    fmedian = np.median(waterfall, axis=1)
+    # Some meta info
+    freqs = uv.freq_array[0, :]
+    pols = uvutils.polnum2str(uv.polarization_array)
+    hera_qm_version = hera_qm_version_str
```

Comment:
Yup, that's already in there. That string has the "version" (currently 0.0.0), git origin, hash, branch, and description.